packages/coordinator/ts/common/accountAbstraction.ts

```diff
@@ -53,3 +55,108 @@ export const getPublicClient = (chainName: ESupportedNetworks): PublicClient<Htt
+ *
+ * @param network - the network we want to interact with
+ * @returns the RPCUrl for the network
+ */
+export const genPimlicoRPCUrl = (network: string): string => {
+  const pimlicoAPIKey = process.env.PIMLICO_API_KEY;
+
+  if (!pimlicoAPIKey) {
+    throw new Error(ErrorCodes.PIMLICO_API_KEY_NOT_SET);
+  }
+
+  return `https://api.pimlico.io/v2/${network}/rpc?apikey=${pimlicoAPIKey}`;
+};
+
+/**
+ * Get the ZeroDev bundler RPC URL based on the network
+ *
+ * @param network - the network we are on
+ * @returns the ZeroDev bundler RPC URL
+ */
+export const getZeroDevBundlerRPCUrl = (network: ESupportedNetworks): string => {
+  switch (network) {
+    case ESupportedNetworks.OPTIMISM_SEPOLIA:
+      return process.env.ZERODEV_BUNDLER_RPC_OP_SEPOLIA || "";
+    case ESupportedNetworks.OPTIMISM:
+      return process.env.ZERODEV_BUNDLER_RPC_OP || "";
+    default:
+      throw new Error(ErrorCodes.UNSUPPORTED_NETWORK);
+  }
+};
+
+/**
+ * Get a bundler client
+ * @param chainName - the chain name
+ * @returns the bundler client
+ */
+export const getBundlerClient = (chainName: ESupportedNetworks): BundlerClient<ENTRYPOINT_ADDRESS_V07_TYPE> =>
+  createBundlerClient({
+    transport: http(genPimlicoRPCUrl(chainName)),
+    chain: viemChain(chainName),
+    entryPoint: ENTRYPOINT_ADDRESS_V07,
+  });
+
+/**
+ * Get a Kernel account handle given a session key
+ * @param sessionKey
+ * @param chainId
+ */
+export const getKernelClient = async (
+  sessionKey: Hex,
+  approval: string,
+  chain: ESupportedNetworks,
+): Promise<
+  KernelAccountClient<
+    ENTRYPOINT_ADDRESS_V07_TYPE,
+    Transport,
+    Chain,
+    KernelSmartAccount<ENTRYPOINT_ADDRESS_V07_TYPE, HttpTransport, Chain>
+  >
+> => {
+  const publicClient = getPublicClient(chain);
+
+  // Using a stored private key
+  const sessionKeySigner = toECDSASigner({
+    signer: privateKeyToAccount(sessionKey),
+  });
+
+  const sessionKeyAccount = await deserializePermissionAccount(
+    publicClient,
+    ENTRYPOINT_ADDRESS_V07,
+    KERNEL_V3_1,
+    approval,
+    sessionKeySigner,
+  );
+
+  const kernelClient = createKernelAccountClient({
+    bundlerTransport: http(process.env.ZERODEV_BUNDLER_RPC),
+    entryPoint: ENTRYPOINT_ADDRESS_V07,
+    account: sessionKeyAccount,
+    chain: viemChain(chain),
+  });
+
+  return kernelClient;
+};
+
+/**
+ * The topic for the contract creation event
+ */
+export const contractCreationEventTopic = "0x4db17dd5e4732fb6da34a148104a592783ca119a1e7bb8829eba6cbadef0b511";
+
+/**
+ * Get the address of the newly deployed contract from a transaction receipt
+ * @param receipt - The transaction receipt
+ * @returns The address of the newly deployed contract
+ */
+export const getDeployedContractAddress = (receipt: TransactionReceipt): string | undefined => {
+  const addr = receipt.logs.find((log) => log.topics[0] === contractCreationEventTopic);
+
+  const deployedAddress = addr ? `0x${addr.topics[1]?.slice(26)}` : undefined;
+
+  return deployedAddress;
```

Comment:
Let's create a const for 26 too
```suggestion
  return addr ? `0x${addr.topics[1]?.slice(26)}` : undefined;
```